### DISCUSSION TITLE
Problem launching the app after exporting on android

### DISCUSSION THREAD

**Discussion by avcr2019** [airsdk/Adobe-Runtime-Support#2636] [General]
Welcome everyone
I wish you all a happy day
Today is the first day I use the ad library for (ANE-Adverts) from distriqt
It works successfully for the Apple system, and the application has been exported, tested, and works successfully
The problem here is in exporting to Android. The application is successfully exported without any error, and when trying and pressing the application to run on the phone, the application never opens. It opens very quickly, less than a second, and does not open. I do not know the reason, knowing that it works successfully on the Apple system and the computer.
I am using the latest harman package which is [AIR 50.2.2.5]
This is the xml file
`<manifest android:installLocation="auto">

			    <uses-sdk android:minSdkVersion="19" android:targetSdkVersion="31"/>

			    
			    <uses-permission android:name="android.permission.INTERNET"/>
			    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
			    <uses-permission android:name="com.google.android.gms.permission.AD_ID" /> 
    
			    
			    <queries>
			        
			        <intent>
			            <action android:name="android.intent.action.VIEW" />
			            <category android:name="android.intent.category.BROWSABLE" />
			            <data android:scheme="https" />
			        </intent>
			        
			        
			        <intent>
			            <action android:name="android.support.customtabs.action.CustomTabsService" />
			        </intent>
			        
			    </queries>
				
			    <uses-permission android:name="android.permission.WAKE_LOCK"/>
			    <uses-permission android:name="com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE"/>
				
				
                <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
				<uses-permission android:name="android.permission.READ_PHONE_STATE"/>
				<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
				<uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
			    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>

			    <application android:hardwareAccelerated="true" android:appComponentFactory="androidx.core.app.CoreComponentFactory">

			        <meta-data
			            android:name="com.google.android.gms.ads.APPLICATION_ID"
			            android:value="ca-app-pub-3940256099942544~3347511713"/>

			        <activity
			            android:name="com.google.android.gms.ads.AdActivity"
			            android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|uiMode|screenSize|smallestScreenSize"
			            android:exported="false"
			            android:theme="@android:style/Theme.Translucent" />

			        <provider
			            android:name="com.google.android.gms.ads.MobileAdsInitProvider"
			            android:authorities="air.English.camp.kids.mobileadsinitprovider"
			            android:exported="false"
			            android:initOrder="100" />

			        <service
			            android:name="com.google.android.gms.ads.AdService"
			            android:enabled="true"
			            android:exported="false" />

			        <activity
			            android:name="com.google.android.gms.common.api.GoogleApiActivity"
			            android:exported="false"
			            android:theme="@android:style/Theme.Translucent.NoTitleBar" />

			        <meta-data
			            android:name="com.google.android.gms.version"
			            android:value="@integer/google_play_services_version" />

			        <provider
			            android:name="androidx.startup.InitializationProvider"
			            android:authorities="air.English.camp.kids.androidx-startup"
			            android:exported="false" >
			            <meta-data
			                android:name="androidx.work.WorkManagerInitializer"
			                android:value="androidx.startup" />
				            <meta-data
				                android:name="androidx.lifecycle.ProcessLifecycleInitializer"
				                android:value="androidx.startup" />
			        </provider>

			        <service
			            android:name="androidx.work.impl.background.systemalarm.SystemAlarmService"
			            android:directBootAware="false"
			            android:enabled="@bool/enable_system_alarm_service_default"
			            android:exported="false" />
			        <service
			            android:name="androidx.work.impl.background.systemjob.SystemJobService"
			            android:directBootAware="false"
			            android:enabled="@bool/enable_system_job_service_default"
			            android:exported="true"
			            android:permission="android.permission.BIND_JOB_SERVICE" />
			        <service
			            android:name="androidx.work.impl.foreground.SystemForegroundService"
			            android:directBootAware="false"
			            android:enabled="@bool/enable_system_foreground_service_default"
			            android:exported="false" />

			        <receiver
			            android:name="androidx.work.impl.utils.ForceStopRunnable$BroadcastReceiver"
			            android:directBootAware="false"
			            android:enabled="true"
			            android:exported="false" />
			        <receiver
			            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$BatteryChargingProxy"
			            android:directBootAware="false"
			            android:enabled="false"
			            android:exported="false" >
			            <intent-filter>
			                <action android:name="android.intent.action.ACTION_POWER_CONNECTED" />
			                <action android:name="android.intent.action.ACTION_POWER_DISCONNECTED" />
			            </intent-filter>
			        </receiver>
			        <receiver
			            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$BatteryNotLowProxy"
			            android:directBootAware="false"
			            android:enabled="false"
			            android:exported="false" >
			            <intent-filter>
			                <action android:name="android.intent.action.BATTERY_OKAY" />
			                <action android:name="android.intent.action.BATTERY_LOW" />
			            </intent-filter>
			        </receiver>
			        <receiver
			            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$StorageNotLowProxy"
			            android:directBootAware="false"
			            android:enabled="false"
			            android:exported="false" >
			            <intent-filter>
			                <action android:name="android.intent.action.DEVICE_STORAGE_LOW" />
			                <action android:name="android.intent.action.DEVICE_STORAGE_OK" />
			            </intent-filter>
			        </receiver>
			        <receiver
			            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$NetworkStateProxy"
			            android:directBootAware="false"
			            android:enabled="false"
			            android:exported="false" >
			            <intent-filter>
			                <action android:name="android.net.conn.CONNECTIVITY_CHANGE" />
			            </intent-filter>
			        </receiver>
			        <receiver
			            android:name="androidx.work.impl.background.systemalarm.RescheduleReceiver"
			            android:directBootAware="false"
			            android:enabled="false"
			            android:exported="false" >
			            <intent-filter>
			                <action android:name="android.intent.action.BOOT_COMPLETED" />
			                <action android:name="android.intent.action.TIME_SET" />
			                <action android:name="android.intent.action.TIMEZONE_CHANGED" />
			            </intent-filter>
			        </receiver>
			        <receiver
			            android:name="androidx.work.impl.background.systemalarm.ConstraintProxyUpdateReceiver"
			            android:directBootAware="false"
			            android:enabled="@bool/enable_system_alarm_service_default"
			            android:exported="false" >
			            <intent-filter>
			                <action android:name="androidx.work.impl.background.systemalarm.UpdateProxies" />
			            </intent-filter>
			        </receiver>
			        <receiver
			            android:name="androidx.work.impl.diagnostics.DiagnosticsReceiver"
			            android:directBootAware="false"
			            android:enabled="true"
			            android:exported="true"
			            android:permission="android.permission.DUMP" >
			            <intent-filter>
			                <action android:name="androidx.work.diagnostics.REQUEST_DIAGNOSTICS" />
			            </intent-filter>
			        </receiver>

			        <service
			            android:name="androidx.room.MultiInstanceInvalidationService"
			            android:directBootAware="true"
			            android:exported="false" />
        
			    </application>

			</manifest>

		]]></manifestAdditions>
        <containsVideo>true</containsVideo>
    </android>
    <versionLabel></versionLabel>
    <extensions>
        <extensionID>com.distriqt.Adverts</extensionID>
        <extensionID>com.distriqt.playservices.AppSet</extensionID>
        <extensionID>com.distriqt.playservices.Base</extensionID>
        <extensionID>com.distriqt.playservices.Ads</extensionID>
        <extensionID>androidx.constraintlayout</extensionID>
        <extensionID>androidx.vectordrawable</extensionID>
        <extensionID>androidx.work</extensionID>
        <extensionID>androidx.room</extensionID>
        <extensionID>com.google.code.gson</extensionID>
        <extensionID>androidx.appcompat</extensionID>
        <extensionID>androidx.browser</extensionID>
        <extensionID>androidx.core</extensionID>
        <extensionID>com.distriqt.Core</extensionID>
    </extensions>
</application>`
I hope you can help me with any solution to this problem. Now I have to upload a new version of the application before it is removed from the store.

**Comment by avcr2019** [comment]
Hello again
I created an empty project and put only buttons in it and experimented to make sure that the problem is not in my application. And the same problem occurred, it works successfully on the iPhone and the computer, and it does not work on Android devices, and I will attach a video to clarify the problem

https://user-images.githubusercontent.com/51138931/235786192-3ae74fc5-2dde-44a7-b56f-6330bd048da6.mp4

**Reply by teotigraphix** [reply]
What are the Logcat logs saying?

Open Android Studio, Logcat tab, clear the messages and open the app.

**Reply by avcr2019** [reply]
> What are the Logcat logs saying?
> 
> Open Android Studio, Logcat tab, clear the messages and open the app.

Thank you for the nice comment
I did not understand what you mean
I am using Adobe Anime 2022 and exporting successfully and I have never encountered this problem. The problem occurred when I used the ad library. It works successfully in Apple and Computer. I will send you a picture of the settings I made in a file (adt.cfg )
While the part about the program (Android Studio) I did not understand and do not know how to do. Please follow up with me to solve the problem, because I want to update the application before deleting it from the store.
Thank you my friend and have a nice day

adt.cfg 

![Screen Shot 2023-05-03 at 2 09 57 AM](https://user-images.githubusercontent.com/51138931/235804390-7e622575-b6d5-4437-904d-191033d9db5b.png)

**Reply by teotigraphix** [reply]
Heh, it looks like you already have Android Studio installed. :) I just noticed that.

IE It's your JAVA_HOME location.

**Comment by teotigraphix** [comment]
1. Download Android Studio - https://developer.android.com/codelabs/basic-android-kotlin-compose-install-android-studio
2. Open this Logcat view - https://developer.android.com/studio/debug/logcat
3. Clear all the output message logs.
4. Run your app with emulator OR USB, Android Studio will pick up your device.
5. Once the app crashes, look in the Logcat window and you should see where it is crashing.
6. Post the logs here I might be able to help.

**Reply by avcr2019** [reply]
Thank you my dear friend for your help
The problem has been solved successfully by adding this library (com.jetbrains.kotlin) with the help of our friend (marchbold )
Thank you for your perseverance and cooperation to solve the problem

**Comment by marchbold** [comment]
Try adding the `com.jetbrains.kotlin` extension, this one is required by the latest androidx libs and missing it would cause a similar crash at startup.

**Reply by avcr2019** [reply]
Thank you my dear friend for your help
This library has already been added and the application works successfully.
It is true that there is a slow start to the application. I don't know what caused it, but it's good that it works and opens.
thanks for the help
